addons/web/controllers/domain.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http, _
+from odoo.http import Controller, request
+from odoo.exceptions import ValidationError
+from odoo.tools.misc import mute_logger
+
+class Domain(Controller):
+
+    @http.route('/web/domain/validate', type='json', auth="user")
+    def validate(self, model, domain):
+        """ Parse `domain` and verify that it can be used to search on `model`
+        :return: True when the domain is valid, otherwise False
+        A validation error is raised if `model` is invalid
+        """
+        try:
+            Model = request.env[model]
+        except Exception:
+            raise ValidationError(_('Invalid model: %s', model))
+        try:
+            # go through the motions of preparing the final SQL for the domain,
+            # so that anything invalid will raise an exception.
+            query = Model._search(domain)
+            sql, params = query.select()
+
+            # Execute the search in EXPLAIN mode, to have the query parser
+            # verify it. EXPLAIN will make sure the query is never actually executed
+            # An alternative to EXPLAIN would be a LIMIT 0 clause, but the semantics
+            # of a falsy `limit` parameter when calling _search() do not permit it.
+            with mute_logger('odoo.sql_db'):
+                request.env.cr.execute(f"EXPLAIN {sql}", params)
+            return True
+        except Exception: # noqa
+            return False
```

Comment:
I think some errors come with a sensitive error message and that it could be interesting to show that error message to the user, at least in debug mode, what do you think?

Reply:
Yes indeed, it could be nice to do that. In the domain field, it is not important right now but it could be in future. DO that in master at some point when needed maybe